tensorflow-framework/src/main/java/org/tensorflow/framework/callbacks/Mode.java

```diff
@@ -0,0 +1,22 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+=======================================================================*/
+package org.tensorflow.framework.callbacks;
+
+/** The mode on when a Callback takes action. */
+public enum Mode {
+  AUTO,
```

Comment:
Some documentation on what `AUTO` means would be good. The others are self explanatory (but should probably have javadoc too if `AUTO` gets it).